src/main/java/net/atos/zac/authentication/UserPrincipalFilter.java

```diff
@@ -92,7 +92,7 @@ public void destroy() {
 
     private LoggedInUser createLoggedInUser(final OidcSecurityContext context) {
         final AccessToken accessToken = context.getToken();
-        final Set<String> roles = Set.copyOf(accessToken.getRealmAccessClaim().getRoles());
+        final Set<String> roles = Set.copyOf(accessToken.getRolesClaim());
```

Comment:
Do we have any test to check the roles are there actually?

Reply:
I don't think there is a direct way to test it in an integration test. We have a IdentityRESTService which returns the logged in user, but it maps it to a RESTLoggedInUser and that doesn't include the roles

Reply:
There is a unit test already :-)
https://github.com/infonl/dimpact-zaakafhandelcomponent/pull/959/files#diff-489d2194c592d8e023e341b4e87e3d52f69d60598d8c1039b4252d7d601d54f4